test/test_profiler.py

```diff
@@ -138,6 +140,182 @@ def test_json_output():
+        uncertainty_per_quarter_second = 0.1
+        absolute_tolerance = (time_in_seconds / 0.25) * uncertainty_per_quarter_second
+        return absolute_tolerance
+
+    speedscope_event_list = speedscope_profile["events"]
+    sleep_a_time_in_seconds = 0.25
+    sleep_b_time_in_seconds = 0.5
+
+    output_event_tuple = (
+        SpeedscopeEvent(
+            SpeedscopeEventType.OPEN,
+            event_time_in_seconds,
+            distinct_functions_called["test_speedscope_output"],
+        ),
+        SpeedscopeEvent(
+            SpeedscopeEventType.OPEN,
+            event_time_in_seconds,
+            distinct_functions_called["long_function_a"],
+        ),
+        SpeedscopeEvent(
+            SpeedscopeEventType.OPEN,
+            event_time_in_seconds,
+            distinct_functions_called["sleep"],
+        ),
+        SpeedscopeEvent(
+            SpeedscopeEventType.CLOSE,
+            update_event_time_on_sleep_close(sleep_a_time_in_seconds),
+            distinct_functions_called["sleep"],
+        ),
+        SpeedscopeEvent(
+            SpeedscopeEventType.CLOSE,
+            event_time_in_seconds,
+            distinct_functions_called["long_function_a"],
+        ),
+        SpeedscopeEvent(
+            SpeedscopeEventType.OPEN,
+            event_time_in_seconds,
+            distinct_functions_called["long_function_b"],
+        ),
+        SpeedscopeEvent(
+            SpeedscopeEventType.OPEN,
+            event_time_in_seconds,
+            distinct_functions_called["sleep"],
+        ),
+        SpeedscopeEvent(
+            SpeedscopeEventType.CLOSE,
+            update_event_time_on_sleep_close(sleep_b_time_in_seconds),
+            distinct_functions_called["sleep"],
+        ),
+        SpeedscopeEvent(
+            SpeedscopeEventType.CLOSE,
+            event_time_in_seconds,
+            distinct_functions_called["long_function_b"],
+        ),
+        SpeedscopeEvent(
+            SpeedscopeEventType.CLOSE,
+            event_time_in_seconds,
+            distinct_functions_called["test_speedscope_output"],
+        ),
+    )
+
+    assert len(speedscope_event_list) == len(output_event_tuple)
+    speedscope_event_fields = tuple([field.name for field in dataclasses.fields(SpeedscopeEvent)])
+    for (event_index, speedscope_event) in enumerate(speedscope_event_list):
+        for event_field in speedscope_event_fields:
+            assert event_field in speedscope_event, (
+                f"Speedscope output 'profiles[0].events[{event_index}]' lacks "
+                f"a field named '{event_field}'"
+            )
+
+        output_event_type = output_event_tuple[event_index].type.value
+        speedscope_event_type = speedscope_event["type"]
+        assert speedscope_event["type"] == output_event_tuple[event_index].type.value, (
+            f"Speedscope output 'profiles[0].events[{event_index}].type has "
+            f"value {speedscope_event_type} != {output_event_type}"
+        )
```

Comment:
I think this section will be quite a lot simplified by removing the assert statements (and associated local variables), as mentioned above.

Reply:
Removed assertion messages and associated local variables in [`97925fb2`](https://github.com/joerick/pyinstrument/pull/160/commits/97925fb284a34a111b7b99bbef8cd0bb72e7c584).